### DISCUSSION TITLE
Any chance supporting Vue will ever be a thing?

### DISCUSSION THREAD

**Discussion by vesper8** [radix-ui/primitives#954] [Ideas] [answered]
Really interested in using this on a client's project, specifically the ScrollArea, but the project is already deep into development and made using Vue. I know this is a React project but, do you think it would be possible to use it in a Vue project?

Thanks

**Comment by jjenzz** [comment]
There are no short term plans to add Vue support as far as I know. We would like to support it (and others) in the future but that could be a good while away I'm afraid.

**Reply by staszek998** [reply]
@jjenzz, I think I'm looking for something to do with my free time... 🤔

I'm pretty experienced with Vue and I'm currently into accessibility and design systems. Having said that, **I thought that rewriting/translating Radix from React to Vue would be an awesomely interesting and growing thing for me**. I should be able to provide the necessary code quality, and at the same time learn a lot from the Radix team. In other words: **I'm positive I'd be a pretty good match for this kind of task**.

What do you think about this?

**Reply by benoitgrelard** [reply]
Hi @staszek998,

That sounds like a great learning experience! 🙂

However I want to emphasise that it isn't a priority for us to work on Vue support which means our capacity for helping on this will be rather limited (as well as our knowledge too as our experience is mostly React).

That said, this absolutely shouldn't stop you from going for it! And the door always remains open for questions, etc.

Good luck! 👍

**Reply by staszek998** [reply]
@benoitgrelard, sadly, at this moment I'm pretty overloaded, but I constantly keep this thread in mind and will come back to it when I'll have some free time. I've already started working, wrote some lines, but I had to put it aside... 😞

**Reply by bryanmylee** [reply]
@staszek998 I'm actually working on a very new project that aims to do what Radix does for all UI libraries! The project is still in the prototyping and design phase as I try to figure out the best way to be UI agnostic, but I've just reached a milestone in making the design state agnostic and reactive.

I'm mostly experienced with React and Svelte so I've only been able to test the design with React's `useState` and Svelte's `writable`, but I'd love your expertise on Vue to discuss some design strategies.

Let me know if you're keen to connect! I hope it's okay if I post my email here, but you can reach me on bryanmylee@gmail.com. If the email goes against the discussion forum's rules I'll redact it.

**Comment by muety** [comment]
I'd love to have Vue support! But I understand that this is far from trivial to implement.

**Comment by JenuelDev** [comment]
Maybe Try https://headlessui.com instead, from tailwind or maybe try [windiCSS](https://windicss.org/) which what I always use on new projects, its fast and easy to use.

**Comment by cn-2k** [comment]
Amazing idea!

**Comment by garrettvercoe** [comment]
I'm thinking as a start to create a npm package for the Radix icons for Vue. Would there be interest in this?
https://www.npmjs.com/package/@radix-ui/react-icons

**Reply by vesper8** [reply]
I continue to be very hopeful and interested in one day using radix-ui with Vue. Currently we have HeadlessUI but I believe radix-ui to be a better version of that and I sure do wish it was possible to use it with Vue!

**Comment by sadeghbarati** [comment, marked as answer]
It's happening thanks to them ❤️ 

- radix-vue
https://github.com/radix-vue/radix-vue

- Oku-ui
https://github.com/oku-ui/primitives

**Reply by renatodeleao** [reply]
Just dropping another library with Vue support that follows with the same accessible/headless components ethos from Radix:
  - `ark-ui` (https://github.com/chakra-ui/ark)

Thanks Radix for leading the way 💚🙏!

**Reply by cn-2k** [reply]
This lib is amazing! Ty 💚

**Reply by sohaha** [reply]
The vue version of ark-ui seems to be missing some documentation
